packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -94,6 +100,8 @@ export interface TokenRatesState extends BaseState {
     Hex,
     Record<string, ContractExchangeRates>
   >;
+  contractPercentChange1d: Record<Hex, ContractExchangeRates>;
+  priceChange1d: Record<Hex, ContractExchangeRates>;
```

Comment:
We'll want to access more of the new `includeMarketData` fields soon.  At least the following, which I think model nicest as:
```
{

    "0xChain1": {
        "0xAddress1": {
            "allTimeHigh": 0,
            "allTimeLow": 0,
            "circulatingSupply": 0,
            "dilutedMarketCap": 0,
            "marketCap": 0,
            "price": 0,
            "priceChange1d": 0,
            "pricePercentChange1d": 0,
            "totalVolume": 0
        },
        ...
    },
    ...
}
```

Rather than:
```
{
    "pricePercentChange1d": {
        "0xChain1": {
            "0xAddress1": 0,
            ...
        }
    },
    "priceChange1d": {
        "0xChain1": {
            "0xAddress1": 0,
            ...
        },
        ...
    },
    ...
}
```

Not sure if we should force old state to migrate, or stick it under a new field for back compat.

Reply:
It looks like https://github.com/MetaMask/core/pull/4206/commits/295320e659f9529097766d416b4dac79b07e4a25 moved to the following structure:

```
{
    "oneDayPriceChange": {
        "0xChain1": {
            "priceChange1d": {
                "0xAddress1": 0
            },
            "pricePercentChange1d": {
                "0xAddress1": 0
            }
        }
    },
}
```

It would be more convenient to get all the market data for a token in 1 object, rather than piece it together from separate objects.  So I still prefer the schema from my original post.  You just index [chain][address] and destructure whatever fields you need.

To support the asset page, we already need to expose many of the other fields we get by requesting `includeMarketData`.  At least the ones in my original post, if you want to handle them in this PR.

Reply:
emmm ok, so you want me to include all the data received from our api call and not just the percentage and amount change ? i'll implement the changes and keep you informed